Local development setup not working

**Describe the issue**
Tried to build a local docker image and it throws up the following error
![image](https://github.com/elyra-ai/elyra/assets/41452484/4f23eb92-fbd0-4e02-9aab-ade989c31ba6)

**To Reproduce**
Steps to reproduce the behavior:
1. Clone repo and cd to the directory
2. Run `make build-server` to create the dist/
3. Run `make kf-notebook-image`
*NOTE* I am using m2 mac and for docker build and run commands, I tried passing both `--platform linux/arm64` and `--platform linux/amd64`.

**Screenshots or log output**
If applicable, add screenshots or log output to help explain your problem.
<details><summary>Log Output</summary>
<pre>
#14 [6/9] RUN jupyter elyra --generate-config
#14 0.895 Traceback (most recent call last):
#14 0.895   File "/opt/conda/bin/jupyter-elyra", line 5, in <module>
#14 0.895     from elyra.elyra_app import launch_instance
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/__init__.py", line 16, in <module>
#14 0.895     from elyra.elyra_app import ElyraApp
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/elyra_app.py", line 35, in <module>
#14 0.895     from elyra.pipeline.handlers import ComponentCacheCatalogHandler
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/pipeline/handlers.py", line 34, in <module>
#14 0.895     from elyra.pipeline.parser import PipelineParser
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/pipeline/parser.py", line 24, in <module>
#14 0.895     from elyra.pipeline.pipeline_definition import Node
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/pipeline/pipeline_definition.py", line 36, in <module>
#14 0.895     from elyra.pipeline.processor import PipelineProcessorManager
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/pipeline/processor.py", line 31, in <module>
#14 0.895     from minio.error import S3Error
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/minio/__init__.py", line 41, in <module>
#14 0.895     from .api import Minio
#14 0.895   File "/opt/conda/lib/python3.8/site-packages/minio/api.py", line 44, in <module>
#14 0.895     from urllib3.response import BaseHTTPResponse
#14 0.895 ImportError: cannot import name 'BaseHTTPResponse' from 'urllib3.response' (/opt/conda/lib/python3.8/site-packages/urllib3/response.py)
#14 ERROR: process "/bin/bash -c jupyter elyra --generate-config" did not complete successfully: exit code: 1
------
 > [6/9] RUN jupyter elyra --generate-config:
0.895     from elyra.pipeline.pipeline_definition import Node
0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/pipeline/pipeline_definition.py", line 36, in <module>
0.895     from elyra.pipeline.processor import PipelineProcessorManager
0.895   File "/opt/conda/lib/python3.8/site-packages/elyra/pipeline/processor.py", line 31, in <module>
0.895     from minio.error import S3Error
0.895   File "/opt/conda/lib/python3.8/site-packages/minio/__init__.py", line 41, in <module>
0.895     from .api import Minio
0.895   File "/opt/conda/lib/python3.8/site-packages/minio/api.py", line 44, in <module>
0.895     from urllib3.response import BaseHTTPResponse
0.895 ImportError: cannot import name 'BaseHTTPResponse' from 'urllib3.response' (/opt/conda/lib/python3.8/site-packages/urllib3/response.py)
------
Dockerfile:43
--------------------
  41 |     
  42 |     # Create and customize Elyra config file to enable only KFP runtime
  43 | >>> RUN jupyter elyra --generate-config
  44 |     RUN sed -i -e "s/# c.PipelineProcessorRegistry.runtimes = \[\]/c.PipelineProcessorRegistry.runtimes = \['kfp'\]/g" $(jupyter --config-dir)/jupyter_elyra_config.py    
  45 |     
--------------------
ERROR: failed to solve: process "/bin/bash -c jupyter elyra --generate-config" did not complete successfully: exit code: 1
make: *** [kf-notebook-image] Error 1
</pre>
</details>

**Expected behavior**
Expected image to build correctly and be able to start a local environment.

**Deployment information**
Describe what you've deployed and how:
N/A.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ dependencies = [
     "jupyterlab-git~=0.32",  # Avoid breaking 1.x changes
     "jupyter-resource-usage>=0.5.1,<1",
     "MarkupSafe>=2.1",
-    "minio>=7.0.0",
+    "minio>=7.0.0,!=7.2.1",
     "nbclient>=0.5.1",
     "nbconvert>=6.5.1",
     "nbdime~=3.1",  # Cap from jupyterlab-git

